src/components/WeatherApp.jsx

```diff
@@ -1,56 +1,110 @@
+    const [isLoading, setIsLoading] = useState(false)
+    const [inputName, setInputName] = useState("Pune");
+    const [saveInputText, setSaveInputText] = useState("Pune");
+    const [error, setError] = useState({ errorText: "", showError: "removeErrorSymbol" })
 
     useEffect(() => {
-        fetch("http://api.weatherapi.com/v1/current.json?key=5bca6bccf3ce4c41a8f15340232807&q=Pune")
+        setIsLoading(true);
+        fetch(`https://api.weatherapi.com/v1/current.json?key=5bca6bccf3ce4c41a8f15340232807&q=${saveInputText}`)
             .then((response) => response.json())
-            .then((data) => console.log(data))
+            .then((data) => {
+                setIsLoading(false)
+                if (data.error) {
+                    setWeatherData(data)
+                    setError({ errorText: "?", showError: "displayErrorSymbol" })
+                    toast.error("Location in valid , please enter a city name.")
+                }
+                else {
+                    setWeatherData(data);
+                    setError({ errorText: "", showError: "removeErrorSymbol" })
+                }
+
+            })
             .catch((error) => console.log(error))
-    }, [])
+    }, [saveInputText])
+    const checkOnKeyDown = (e) => {
+        if (e.key === "Enter") {
+            setSaveInputText(inputName)
+        }
+    }
+    const onClickRenderWeather = () => {
+        if (inputName !== "")
+            setSaveInputText(inputName)
+    }
     return (
         <div >
             <main>
                 <div className="weatherApp" >
                     <div className="searchSection">
-                        <input type="text" className='inputText' placeholder='Pune' />
+                        <input type="text" className='inputText' placeholder='Search...' value={inputName} onChange={(e) => setInputName(e.target.value)} onKeyDown={checkOnKeyDown} />
                         <div className='searchIcon'>
-                            <FiSearch size={19} />
+                            <FiSearch onClick={onClickRenderWeather} />
                         </div>
                     </div>
                     <div >
                         <img src="https://previews.123rf.com/images/martialred/martialred1702/martialred170200035/71710170-cloudy-or-cloud-partly-blocking-the-sun-flat-color-vector-icon-for-weather-apps-and-websites.jpg" alt="weather Img" className=' weatherImg' />
                     </div>
-                    <div>
-                        <div className="temperature ">
-                            <h2>28 &#8451;</h2>
-                            <p>Scattered Clouds</p>
+                    <div  >
+                        <div className='subMainSection'>
+                            <div className="tempMainDiv ">
+                                <div className='tempSubDiv'>
+                                    <h2 className={`h2-${error.showError}`}>{isLoading ? <BeatLoader
```

Comment:
i didnt understand how is this error.showError working? 
why not do it like
className = { error ? "h2-YourErrorClassname" : "h2 } 
much more readable right?

Reply:
yes.